lessons/misc/intro-git/lesson.md

```diff
@@ -0,0 +1,394 @@
+
+Make sure to _NOT_ check the 'Initialize this repository with a README' box at
+the bottom of the page. This is used when creating an entirely new repo on
+GitHub itself, and not when importing an existing local repo.
+
+Copy the link shown at the top of the page to your clipboard,
+and head back to bash. 
+
+```bash
+pwd # make sure we're in the repo directory
+git remote add origin [link]
+git push -u origin master
+```
+
+Here, `origin` refers to the GitHub version of our repo, while `master` refers
+to the branch currently being worked in. We won't worry too much about branches
+for the purpose of this lesson -- that'll be covered in more depth when we
+discuss collaboration with GitHub. 
+
+Let's now refresh the GitHub page. `bio.txt` now appears on GitHub!  Not only
+that, but the latest commit message is listed just above the file tree.
+
+## Pushing and pulling
+
+Click on `bio.txt` in the file tree to open it on GitHub. On the right hand
+side of the page are a few icons; let's click on the pencil (edit) icon to make
+some new changes. Underneath the existing info about yourself, list your
+favourite colour. Once done, head to the bottom of the page to add a message
+and commit these changes.
+
+Great -- let's now head back to the command line and look at the local copy of `bio.txt`.
+
+```bash
+cat bio.txt
+```
+
+The edit we just made on GitHub doesn't seem to be there! 
+
+To sync our local repo with the GitHub repo, we have to _pull_
+the changes that are currently on GitHub. 
+
+```bash
+git pull
+```
+
+This will download the changes we made on GitHub earlier.
+
+Conversely, if changes are made locally, we would have to _push_ to GitHub.
+Let's do just that with `bio.txt`. Open the file and add the brand of computer
+you have at the bottom of the file.
+
+Before we can push our changes, we have to go through the motions of committing
+them first:
+
+```bash
+git add bio.txt
+git commit -m 'added laptop brand'
+```
+
+and then we can push our changes. When pushing, remember that we
+have to specify these changes are heading to the GitHub repo (`origin`)
+and that they're going to the master branch for now:
+
+```bash
+git push origin master
+```
+
+And now we're all synced up on GitHub!
+
+# Wrapping up
+
+We've covered some really important fundamentals for working with Git and GitHub today:
+
+- Command line Git and its core commands
+    - `init`
+    - `add`
+    - `commit`
+    - `diff`
+    - `status`
+    - `log`
+- Syncing a repository on GitHub
+    - `remote`
+    - `push`
+    - `pull`
+
+These are sufficient for getting started with Git and maintaining a personal
+codebase in a given repo. Although this is just one relatively simple use case
+for Git and GitHub, it is incredibly valuable to version control your code and
+make sure all the changes that have been made over time are documented and
+accounted for. It also means that removing/deleting code (accidentally or
```

Comment:
Maybe mention that you can version control any type of file (e.g. manuscript file, data). Of course, it's most helpful for flat text files.